app/code/Magento/Config/Block/System/Config/Form.php

```diff
@@ -366,9 +366,12 @@ protected function _initElement(
 
         $sharedClass = $this->_getSharedCssClass($field);
         $requiresClass = $this->_getRequiresCssClass($field, $fieldPrefix);
+        $isReadOnly = $this->isDefaultFieldReadOnly($path);
+        if (!$isReadOnly) {
+            $isReadOnly = $this->getElementVisibility()->isDisabled($field->getPath())
+                ?: $this->getSettingChecker()->isReadOnly($path, $this->getScope(), $this->getStringScopeCode());
```

Comment:
Please consider injecting `\Magento\Config\Model\Config\Reader\Source\Deployed\SettingChecker`as `\Magento\Theme\Model\Design\Config\DataProvider::getSettingChecker` method is deprecated

Reply:
@sivaschenko do i need to inject `\Magento\Config\Model\Config\Reader\Source\Deployed\SettingChecker` in constructor?

Reply:
Hi @maheshWebkul721 , yes, please